XenModel/Actions/Updates/CheckForCdnUpdatesAction.cs

```diff
@@ -67,15 +78,39 @@ protected override void Run()
                 Query = $"session_id={Uri.EscapeDataString(Session.opaque_ref)}"
             };
 
-            string json;
+            try
+            {
+                string json;
+
+                using (Stream httpStream = HTTPHelper.GET(builder.Uri, Connection, true))
+                {
+                    using (var streamReader = new StreamReader(httpStream))
+                        json = streamReader.ReadToEnd();
+                }
 
-            using (Stream httpStream = HTTPHelper.GET(builder.Uri, Connection, true))
+                Updates = JsonConvert.DeserializeObject<CdnPoolUpdateInfo>(json);
+            }
+            catch (HTTP.BadServerResponseException ex)
             {
-                using (var streamReader = new StreamReader(httpStream))
-                    json = streamReader.ReadToEnd();
+                if (ex.Message.Contains("Received error code HTTP/1.1 404 Not Found\r\n from the server"))
+                {
+                    log.Warn("Failed to retrieve available updates.");
+                    log.Warn(ex.Message);
+                    log.Warn("One of the following errors has occurred: " +
+                             "no_repository_enabled or invalid_updateinfo_xml or invalid_repomd_xml. " +
+                             "See the server side logs for details.");
+                }
+                else if (ex.Message.Contains("Received error code HTTP/1.1 500 Internal Server Error\r\n from the server"))
+                {
+                    log.Warn("Failed to retrieve available updates.");
+                    log.Warn(ex.Message);
+                    log.Warn("One of the following errors has occurred: " +
+                             "get_host_updates_failed or updates_require_sync. " +
+                             "See the server side logs for details.");

```

Comment:
should we avoid listing errors? they could change in the future. I'd just leave "see the server side logs for details". same for 404s above

Reply:
I have raised CP-44822 for a proper fix, so I epxect this to change aagian at some point. Until then, I only listed them lest someone complains the application logs show nothing. If you think it is not necessary, I can remove them, up to you.

Reply:
>  I only listed them lest someone complains the application logs show nothing.

I think the warnings in general are very helpful and we should have them in the XC logs. I just want to remove the list of possible errors. They could become outdated and there's no way we'll remember to update them. Let's just point people to xensource.log